pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -84,7 +92,17 @@ func ApplyFeatureGates() {
 		factory.InsertPredicateKeyToAlgorithmProviderMap(predicates.PodToleratesNodeTaintsPred)
 		factory.InsertPredicateKeyToAlgorithmProviderMap(predicates.CheckNodeUnschedulablePred)
 
-		klog.Infof("TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory")
+		neededPredicates = sets.NewString(predicates.PodToleratesNodeTaintsPred)
```

Comment:
why wasn't CheckNodeUnschedulablePred added in the neededPredicates?

Reply:
@ravisantoshgudimetla I pulled this from your commit, can you explain?

Reply:
Didn't know if this was intentional or not, but it seems it wasn't. @ahg-g updated, thanks

Reply:
Ahh sorry for the delay, @damemi pinged me today to look into it. Now, I realized that at that time, we were thinking of deprecating that Predicate altogether, so I did not add it but we did not go that route, so yes, it is needed

https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/predicates/predicates.go#L1527